### PR TITLE
evaluating Rstr in globalEnv

### PR DESCRIPTION
**Pull request by randy3k**
Improvement to handle the variable substitution, but it seems that there is an memory leak issue. https://github.com/JuliaStats/RCall.jl/issues/123

### CODE REVIEW

**Comment by randy3k** [comment]
I guess I have figured out the memory leak issue. 

Consider
```julia
macro foo()
    quote
        ret = reval("1:10^8")
        ret
    end
end
```

```
julia> @foo;

julia> gc()

R> gc()
           used  (Mb) gc trigger  (Mb) max used  (Mb)
Ncells   218360  11.7     460000  24.6   350000  18.7
Vcells 50323035 384.0   72867244 556.0 50324576 384.0
```

**Comment by randy3k** [comment]
I have posted an issue on Julia https://github.com/JuliaLang/julia/issues/17044.

**Comment by randy3k** [comment]
@simonbyrne I have added the protect and unprotect pair for `env`.

**Comment by randy3k** [comment]
It turns out we should use a `let` block to prevent variables to be exported to global env.

File: src/rstr.jl [hunk 1]
```diff
@@ -103,24 +102,27 @@ All such Julia expressions are evaluated once, before the R expression is evalua
...
         push!(blk_ld.args,:(env[$rsym] = $(esc(expr))))
     end
     quote
-        env = newEnvironment()
-        protect(env)
-        $blk_ld
-        ret = reval($script, env)
-        unprotect(1)
-        ret
+        let
```
Comment (yuyichao) [inline src/rstr.jl]:
```jl
let ret, env = protect(newEnvironment())
    .....
end
```

Comment (randy3k) [inline src/rstr.jl]: thanks

**Pull request merged**